In [57]:
%pylab inline

Populating the interactive namespace from numpy and matplotlib


In [58]:
Ionpos_Methanediol_ang_nc=[[ 1.35170 ,  -0.08370,    0.00000  ], 
                    [-0.52950,    1.24650 ,  -0.00000  ], 
                    [-0.06950,   -0.09830,   -0.00000  ],
                    [-0.43450 ,  -0.61450  ,  0.88910  ],
                    [-0.43450 ,  -0.61450 ,  -0.88910 ],
                    [1.62470 ,  -1.02540  ,  0.00000  ],
                    [-1.50830 ,   1.19000  ,  0.00000]]
ions_Methanediol=['O','O','C','H','H','H','H']

In [59]:
Ionpos_Methanediol_ang_nr= Ionpos_Methanediol_ang_nc-np.array([-0.43450 ,  -0.61450 ,  -0.88910])  #shift H that doesnt belong to the origin
print(Ionpos_Methanediol_ang_nr)

[[ 1.7862  0.5308  0.8891]
 [-0.095   1.861   0.8891]
 [ 0.365   0.5162  0.8891]
 [ 0.      0.      1.7782]
 [ 0.      0.      0.    ]
 [ 2.0592 -0.4109  0.8891]
 [-1.0738  1.8045  0.8891]]


In [60]:
b=np.mat([0,0,1])
a=np.mat([ 0.365 ,  0.5162 , 0.8891]) #rotate to make z axis the one connecting the hydrogen to be removed and the carbon
a=a/np.sqrt(np.dot(a,a.T))
c=np.dot(a,b.T)
v=np.cross(a,b)[0]
s=np.sqrt(np.dot(v,v.T))
y = np.mat([[0, -v[2], v[1]], [v[2], 0, -v[0]], [-v[1], v[0], 0]])

Rot_Methanediol=np.mat(np.diag([1,1,1]))+y+np.dot(y,y)*(((1-c)/s**2)[0,0])
Ionpos_Methanediol_ang=np.array(np.dot(Rot_Methanediol,np.mat(Ionpos_Methanediol_ang_nr.T))).T

In [61]:
 Ionpos_Formaldehyde_ang_nc=np.array([[ 0.000000   ,  .000000   ,  .683501],
            [.000000   ,  .000000 ,   -.536614],
            [.000000  ,   .934390  , -1.124164],
            [.000000  ,  -.934390  , -1.124164]])
ions_Formaldehyde=['O','C','H','H']

In [62]:
Ionpos_Formaldehyde_ang_nr= Ionpos_Formaldehyde_ang_nc-np.array([.000000  ,  -.934390  , -1.124164]) #shift H that doesnt belong to the origin
print(Ionpos_Formaldehyde_ang_nr)

[[0.       0.93439  1.807665]
 [0.       0.93439  0.58755 ]
 [0.       1.86878  0.      ]
 [0.       0.       0.      ]]


In [63]:
b=np.mat([0,0,1])
a=np.mat([0.     ,  0.93439 , 0.58755 ]) #rotate to make z axis the one connecting the hydrogen to be removed and the carbon
a=a/np.sqrt(np.dot(a,a.T))
c=np.dot(a,b.T)
v=np.cross(a,b)[0]
s=np.sqrt(np.dot(v,v.T))
y = np.mat([[0, -v[2], v[1]], [v[2], 0, -v[0]], [-v[1], v[0], 0]])

Rot_Formaldehyde=np.mat(np.diag([1,1,1]))+y+np.dot(y,y)*(((1-c)/s**2)[0,0])
Ionpos_Formaldehyde_ang=np.array(np.dot(Rot_Formaldehyde,np.mat(Ionpos_Formaldehyde_ang_nr.T))).T


In [64]:
Ionpos_CO2_ang_nr=np.array([[ 0.000000  ,  0.000000 ,0.000000],
                            [0.00000, 0.000000   , 1.159076],
                            [ 0.000000   , 0.000000  , -1.159076]])
ions_CO2=['C','O','O']

In [65]:
b=np.mat([1,0,0])
a=np.mat([0.     ,  0 , 1 ])
a=a/np.sqrt(np.dot(a,a.T))
c=np.dot(a,b.T)
v=np.cross(a,b)[0]
s=np.sqrt(np.dot(v,v.T))
y = np.mat([[0, -v[2], v[1]], [v[2], 0, -v[0]], [-v[1], v[0], 0]])

Rot_CO2=np.mat(np.diag([1,1,1]))+y+np.dot(y,y)*(((1-c)/s**2)[0,0])
Ionpos_CO2_ang=np.array(np.dot(Rot_CO2,np.mat(Ionpos_CO2_ang_nr.T))).T


In [66]:
Ionpos_Formic_acid_ang_nc=np.array([[0.7910  ,   -0.0921   ,   1.3425],
   [0.3208  ,   -0.0411   ,   0.3549],
   [0.8042   ,  -0.1358   ,  -0.7518],
   [-1.0065  ,    0.1597   ,   0.4283],
   [-1.4043   ,   0.2023  ,   -0.4370]])  
ions_Formic_acid=['H','C','O','O','H']

In [67]:
b=np.mat([0,0,1])
a=np.mat(np.cross(Ionpos_Formic_acid_ang_nc[1]-Ionpos_Formic_acid_ang_nc[0], Ionpos_Formic_acid_ang_nc[2]-Ionpos_Formic_acid_ang_nc[0])) #the molecule is planar, we rotate to set the normal vector of the plane to line up with the z axis
a=a/np.sqrt(np.dot(a,a.T))
c=np.dot(a,b.T)
v=np.cross(a,b)[0]
s=np.sqrt(np.dot(v,v.T))
y = np.mat([[0, -v[2], v[1]], [v[2], 0, -v[0]], [-v[1], v[0], 0]])

Rot_Formic_acid=np.mat(np.diag([1,1,1]))+y+np.dot(y,y)*(((1-c)/s**2)[0,0])
Ionpos_Formic_acid_ang=np.array(np.dot(Rot_Formic_acid,np.mat(Ionpos_Formic_acid_ang_nc.T))).T


In [68]:
#attempt at creating the Pt file
Coords1=(np.linspace(0,0.5,2))/2
Coords2=(np.linspace(0,0.5,2))/2
Coords3=(np.linspace(-0.2,.2,5))
print(Coords1,Coords2,Coords3)

[0.   0.25] [0.   0.25] [-0.2 -0.1  0.   0.1  0.2]


In [69]:
Ionpos_Pt=[[i,j,k] for i in Coords1 for j in Coords2 for k in Coords3]
ions_Pt=['Pt' for i in range(len(Ionpos_Pt))]

In [70]:
for i in range(len(Ionpos_Pt)):
    print('Ion '+ions_Pt[i], end=' ' )
    for j in range(len(Ionpos_Pt[i])):
        print("%.3f" % Ionpos_Pt[i][j], end = ' ')
    print('\n')

Ion Pt 0.000 0.000 -0.200 

Ion Pt 0.000 0.000 -0.100 

Ion Pt 0.000 0.000 0.000 

Ion Pt 0.000 0.000 0.100 

Ion Pt 0.000 0.000 0.200 

Ion Pt 0.000 0.250 -0.200 

Ion Pt 0.000 0.250 -0.100 

Ion Pt 0.000 0.250 0.000 

Ion Pt 0.000 0.250 0.100 

Ion Pt 0.000 0.250 0.200 

Ion Pt 0.250 0.000 -0.200 

Ion Pt 0.250 0.000 -0.100 

Ion Pt 0.250 0.000 0.000 

Ion Pt 0.250 0.000 0.100 

Ion Pt 0.250 0.000 0.200 

Ion Pt 0.250 0.250 -0.200 

Ion Pt 0.250 0.250 -0.100 

Ion Pt 0.250 0.250 0.000 

Ion Pt 0.250 0.250 0.100 

Ion Pt 0.250 0.250 0.200 



In [71]:
Coords1=(np.linspace(0,0.5,2))
Coords2=(np.linspace(0,0.5,2))
Coords3=(np.linspace(-0.2,.2,5))
print(Coords1,Coords2,Coords3)

[0.  0.5] [0.  0.5] [-0.2 -0.1  0.   0.1  0.2]


In [72]:
Ionpos_Pt=[[i,j,k] for i in Coords1 for j in Coords2 for k in Coords3]
ions_Pt=['Pt' for i in range(len(Ionpos_Pt))]

In [73]:
for i in range(len(Ionpos_Pt)):
    print('Ion '+ions_Pt[i], end=' ' )
    for j in range(len(Ionpos_Pt[i])):
        print("%.3f" % Ionpos_Pt[i][j], end = ' ')
    print('\n')

Ion Pt 0.000 0.000 -0.200 

Ion Pt 0.000 0.000 -0.100 

Ion Pt 0.000 0.000 0.000 

Ion Pt 0.000 0.000 0.100 

Ion Pt 0.000 0.000 0.200 

Ion Pt 0.000 0.500 -0.200 

Ion Pt 0.000 0.500 -0.100 

Ion Pt 0.000 0.500 0.000 

Ion Pt 0.000 0.500 0.100 

Ion Pt 0.000 0.500 0.200 

Ion Pt 0.500 0.000 -0.200 

Ion Pt 0.500 0.000 -0.100 

Ion Pt 0.500 0.000 0.000 

Ion Pt 0.500 0.000 0.100 

Ion Pt 0.500 0.000 0.200 

Ion Pt 0.500 0.500 -0.200 

Ion Pt 0.500 0.500 -0.100 

Ion Pt 0.500 0.500 0.000 

Ion Pt 0.500 0.500 0.100 

Ion Pt 0.500 0.500 0.200 



In [74]:
for i in range(len(Ionpos_Formaldehyde_ang)):
    print('Ion '+ions_Formaldehyde[i], end=' ' )
    for j in range(len(Ionpos_Formaldehyde_ang[i])):
        print("%.3f" % Ionpos_Formaldehyde_ang[i][j], end = ' ')
    print('\n')

Ion O 0.000 -1.033 1.753 

Ion C 0.000 0.000 1.104 

Ion H 0.000 0.995 1.582 

Ion H 0.000 0.000 0.000 



In [75]:
for i in range(len(Ionpos_Methanediol_ang)):
    print('Ion '+ions_Methanediol[i], end=' ' )
    for j in range(len(Ionpos_Methanediol_ang[i])):
        print("%.3f" % Ionpos_Methanediol_ang[i][j], end = ' ')
    print('\n')

Ion O 1.332 -0.111 1.573 

Ion O -0.549 1.219 1.573 

Ion C 0.000 0.000 1.091 

Ion H -0.595 -0.841 1.449 

Ion H 0.000 0.000 0.000 

Ion H 1.671 -0.961 1.219 

Ion H -1.462 1.255 1.219 



In [76]:
for i in range(len(Ionpos_Formic_acid_ang)):
    print('Ion '+ions_Formic_acid[i], end=' ' )
    for j in range(len(Ionpos_Formic_acid_ang[i])):
        print("%.3f" % Ionpos_Formic_acid_ang[i][j], end = ' ')
    print('\n')


Ion H 0.987 1.210 -0.000 

Ion C 0.373 0.303 -0.000 

Ion O 0.695 -0.865 -0.000 

Ion O -0.944 0.575 0.000 

Ion H -1.468 -0.221 0.000 



In [77]:

for i in range(len(Ionpos_CO2_ang)):
    print('Ion '+ions_CO2[i], end=' ' )
    for j in range(len(Ionpos_CO2_ang[i])):
        print("%.3f" % Ionpos_CO2_ang[i][j], end = ' ')
    print('\n')

Ion C 0.000 0.000 0.000 

Ion O 1.159 0.000 0.000 

Ion O -1.159 0.000 0.000 



In [78]:
Ionpos_Formaldehyde_p=np.array(Ionpos_Formaldehyde_ang)-array(Ionpos_Formaldehyde_ang[-1])
Ionpos_Methanediol_p=np.array(Ionpos_Methanediol_ang)-array(Ionpos_Methanediol_ang[4])
Ionpos_Formic_acid_p=np.array(Ionpos_Formic_acid_ang)-array(Ionpos_Formic_acid_ang[1])
Ionpos_CO2_p=np.array(Ionpos_CO2_ang)
for i in range(len(Ionpos_Formaldehyde_p)):
    print('Ion '+ions_Formaldehyde[i], end=' ' )
    for j in range(len(Ionpos_Formaldehyde_p[i])):
        print("%.3f" % Ionpos_Formaldehyde_p[i][j], end = ' ')
    print('\n')
print('__________________________')
for i in range(len(Ionpos_Methanediol_p)):
    print('Ion '+ions_Methanediol[i], end=' ' )
    for j in range(len(Ionpos_Methanediol_p[i])):
        print("%.3f" % Ionpos_Methanediol_p[i][j], end = ' ')
    print('\n')
    
print('__________________________')
for i in range(len(Ionpos_Formic_acid_p)):
    print('Ion '+ions_Formic_acid[i], end=' ' )
    for j in range(len(Ionpos_Formic_acid_p[i])):
        print("%.3f" % Ionpos_Formic_acid_p[i][j], end = ' ')
    print('\n')
print('__________________________')
for i in range(len(Ionpos_CO2_p)):
    print('Ion '+ions_CO2[i], end=' ' )
    for j in range(len(Ionpos_CO2_p[i])):
        print("%.3f" % Ionpos_CO2_p[i][j], end = ' ')
    print('\n')

Ion O 0.000 -1.033 1.753 

Ion C 0.000 0.000 1.104 

Ion H 0.000 0.995 1.582 

Ion H 0.000 0.000 0.000 

__________________________
Ion O 1.332 -0.111 1.573 

Ion O -0.549 1.219 1.573 

Ion C 0.000 0.000 1.091 

Ion H -0.595 -0.841 1.449 

Ion H 0.000 0.000 0.000 

Ion H 1.671 -0.961 1.219 

Ion H -1.462 1.255 1.219 

__________________________
Ion H 0.614 0.907 0.000 

Ion C 0.000 0.000 0.000 

Ion O 0.322 -1.168 0.000 

Ion O -1.317 0.272 0.000 

Ion H -1.840 -0.524 0.000 

__________________________
Ion C 0.000 0.000 0.000 

Ion O 1.159 0.000 0.000 

Ion O -1.159 0.000 0.000 



In [79]:
Ionpos_Formaldehyde_p=Ionpos_Formaldehyde_p #2.05 is the bond length of carbon and Pt
Ionpos_Methanediol_p=Ionpos_Methanediol_p

In [80]:
Ionpos_Formaldehyde_p=Ionpos_Formaldehyde_p #2.05 is the bond length of carbon and Pt
Ionpos_Methanediol_p=Ionpos_Methanediol_p

In [81]:
#conversion factors for orthorhombic lattice
a_Pt_pre1=0.392 #nm
a_Pt_pre2=0.392 #nm
a_Pt_pre3=0.392 #nm
a_Pt_ang=a_Pt*10 #angstrom
a_Pt_ang1=2*a_Pt_ang/sqrt(2) #angstrom
a_Pt_ang2=2*a_Pt_ang/sqrt(2/3) #angstrom
a_Pt_ang3=5*(sqrt(3)*a_Pt_ang/3)+5*(sqrt(3)*a_Pt_ang/3)  #angstrom (each layer is sqrt(3)a/3 ) 5 layers of material and 5 of separation


#conversion factors for orthorhombic lattice
a_Pt=0.392 #nm
a_Pt_ang=a_Pt*10 #angstrom
a_Pt_ang1=2*a_Pt_ang/sqrt(2) #angstrom
a_Pt_ang2=2*a_Pt_ang/sqrt(2) #angstrom
a_Pt_ang3=5*(sqrt(3)*a_Pt_ang/3)+5*(sqrt(3)*a_Pt_ang/3)  #angstrom (each layer is sqrt(3)a/3 ) 5 layers of material and 5 of separation


print(a_Pt_ang3,a_Pt_ang1*1.88973,a_Pt_ang2*1.88973,a_Pt_ang3*1.88973)
#A100_Formaldehyde=np.array([np.ones(len(Ionpos_Formaldehyde_p))*a_Pt_ang1,np.ones(len(Ionpos_Formaldehyde_p))*a_Pt_ang2,np.ones(len(Ionpos_Formaldehyde_p))*a_Pt_ang3])
#A100_Ionpos_Methanediol=np.array([np.ones(len(Ionpos_Methanediol_p))*a_Pt_ang1,np.ones(len(Ionpos_Methanediol_p))*a_Pt_ang2,np.ones(len(Ionpos_Methanediol_p))*a_Pt_ang3])



### change of basis from rotated hexagonal to euclidean (this is just stacking the basis vectors of the hexagonal 
#### side by side)
A100_CHangebasis_inv=(mat(array([[0 , -a_Pt_ang/2 , a_Pt_ang],[a_Pt_ang/2  , 0 ,  a_Pt_ang],[-a_Pt_ang/2 , a_Pt_ang/2 , a_Pt_ang]])))
###rotating the vectors found in the tutorial (rotated hexagonal that points in the 111 direction) 
### to get the hexagonal lattice vectors perpendicular to the z axis to setup the molecule coordinates

b=np.mat([0,0,1])
a=np.mat([1 ,  1 , 1 ]) #rotate to make z axis the 111 direction
a=a/np.sqrt(np.dot(a,a.T))
c=np.dot(a,b.T)
v=np.cross(a,b)[0]
s=np.sqrt(np.dot(v,v.T))
y = np.mat([[0, -v[2], v[1]], [v[2], 0, -v[0]], [-v[1], v[0], 0]])

Rot=np.mat(np.diag([1,1,1]))+y+np.dot(y,y)*(((1-c)/s**2)[0,0]) ## rotates to hexagonal with z axis as the c axis
A100_CHangebasis_inv_True=np.array(np.dot(Rot,A100_CHangebasis_inv)) ### change of basis from hexagonal to euclidean

##scaling the coordinates
A100_CHangebasis_inv_scaled=dot(diag([2,2,a_Pt_ang3/A100_CHangebasis_inv_True[2][2]]),A100_CHangebasis_inv_True)
A100_CHangebasis=inv(A100_CHangebasis_inv_scaled) ##change of basis from  euclidean to hexagonal


22.632130552233328 10.47612863727537 10.47612863727537 42.768616068471886


In [82]:
Ionpos_Formaldehyde_pp=array((dot(A100_CHangebasis,Ionpos_Formaldehyde_p.T)).T)

In [83]:
Ionpos_Methanediol_pp=array((dot(A100_CHangebasis,Ionpos_Methanediol_p.T)).T)

In [84]:
Ionpos_Formic_acid_pp=array((dot(A100_CHangebasis,Ionpos_Formic_acid_p.T)).T)

In [85]:
Ionpos_CO2_pp=array((dot(A100_CHangebasis,Ionpos_CO2_p.T)).T)

In [86]:
for i in range(len(Ionpos_Formaldehyde_pp)):
    print('ion '+ions_Formaldehyde[i], end=' ' )
    for j in range(len(Ionpos_Formaldehyde_pp[i])):
        print("%.3f" % Ionpos_Formaldehyde_pp[i][j], end = ' ')
    print('1')
print('__________________________')
for i in range(len(Ionpos_Methanediol_pp)):
    print('ion '+ions_Methanediol[i], end=' ' )
    for j in range(len(Ionpos_Methanediol_pp[i])):
        print("%.3f" % Ionpos_Methanediol_pp[i][j], end = ' ')
    print('1')
print('__________________________')    
for i in range(len(Ionpos_Formic_acid_pp)):
    print('ion '+ions_Formic_acid[i], end=' ' )
    for j in range(len(Ionpos_Formic_acid_pp[i])):
        print("%.3f" % Ionpos_Formic_acid_pp[i][j], end = ' ')
    print('1')
print('__________________________') 
for i in range(len(Ionpos_CO2_pp)):
    print('ion '+ions_CO2[i], end=' ' )
    for j in range(len(Ionpos_CO2_pp[i])):
        print("%.3f" % Ionpos_CO2_pp[i][j], end = ' ')
    print('1')

ion O -0.208 -0.056 0.077 1
ion C 0.000 -0.000 0.049 1
ion H 0.200 0.054 0.070 1
ion H 0.000 0.000 0.000 1
__________________________
ion O -0.094 -0.274 0.070 1
ion O 0.275 0.176 0.070 1
ion C 0.000 -0.000 0.048 1
ion H -0.137 0.074 0.064 1
ion H 0.000 0.000 0.000 1
ion H -0.283 -0.388 0.054 1
ion H 0.331 0.362 0.054 1
__________________________
ion H 0.149 -0.075 0.000 1
ion C 0.000 0.000 0.000 1
ion O -0.252 -0.128 0.000 1
ion O 0.126 0.280 0.000 1
ion H -0.006 0.342 0.000 1
__________________________
ion C 0.000 0.000 0.000 1
ion O -0.062 -0.233 0.000 1
ion O 0.062 0.233 0.000 1


In [87]:
floor((5*(sqrt(3)*a_Pt_ang/3))*1.88973 +15)

36.0

In [88]:
#distance to platinum
Dis_ang=.186*10 #literature value in angstrom
Dis=Dis_ang/a_Pt_ang3 #in fractional coordinates 


#summing the position of the central atom in the top most layer of  the surface

Dis

0.08218404342036

In [89]:
Ionpos_Formaldehyde_ppp=Ionpos_Formaldehyde_pp+np.array([0,0,Dis-0.049])+np.array([0.5,0.5, 0.2001]) #substracting top carbon pos after rescaling to 6 separation layers then adding top PT and bond distance from literature
Ionpos_Methanediol_ppp=Ionpos_Methanediol_pp+np.array([0,0,Dis-0.049  ])+np.array([0.5,0.5, 0.2001]) #substracting top carbon pos then adding top PT and bond distance from literature
Ionpos_Formic_acid_ppp=Ionpos_Formic_acid_pp+np.array([0,0,Dis ])+np.array([0.5,0.5,1.4*0.2001])
Ionpos_CO2_ppp=Ionpos_CO2_pp+np.array([0,0,Dis ])+np.array([0.5,0.5,1.4*0.2001])

In [90]:
for i in range(len(Ionpos_Formaldehyde_ppp)):
    print('ion '+ions_Formaldehyde[i], end=' ' )
    for j in range(len(Ionpos_Formaldehyde_ppp[i])):
        print("%.3f" % Ionpos_Formaldehyde_ppp[i][j], end = ' ')
    print(' 1')
print('__________________________')
for i in range(len(Ionpos_Methanediol_ppp)):
    print('ion '+ions_Methanediol[i], end=' ' )
    for j in range(len(Ionpos_Methanediol_ppp[i])):
        print("%.3f" % Ionpos_Methanediol_ppp[i][j], end = ' ')
    print(' 1')
print('__________________________')
for i in range(len(Ionpos_Formic_acid_ppp)):
    print('ion '+ions_Formic_acid[i], end=' ' )
    for j in range(len(Ionpos_Formic_acid_ppp[i])):
        print("%.3f" % Ionpos_Formic_acid_ppp[i][j], end = ' ')
    print(' 1')
print('__________________________')
for i in range(len(Ionpos_CO2_ppp)):
    print('ion '+ions_CO2[i], end=' ' )
    for j in range(len(Ionpos_CO2_ppp[i])):
        print("%.3f" % Ionpos_CO2_ppp[i][j], end = ' ')
    print(' 1') 

ion O 0.292 0.444 0.311  1
ion C 0.500 0.500 0.282  1
ion H 0.700 0.554 0.303  1
ion H 0.500 0.500 0.233  1
__________________________
ion O 0.406 0.226 0.303  1
ion O 0.775 0.676 0.303  1
ion C 0.500 0.500 0.281  1
ion H 0.363 0.574 0.297  1
ion H 0.500 0.500 0.233  1
ion H 0.217 0.112 0.287  1
ion H 0.831 0.862 0.287  1
__________________________
ion H 0.649 0.425 0.362  1
ion C 0.500 0.500 0.362  1
ion O 0.248 0.372 0.362  1
ion O 0.626 0.780 0.362  1
ion H 0.494 0.842 0.362  1
__________________________
ion C 0.500 0.500 0.362  1
ion O 0.438 0.267 0.362  1
ion O 0.562 0.733 0.362  1


ion Pt 0.000 0.000 -0.200 1
ion Pt 0.500 0.500 -0.200 1
ion Pt 0.000 0.50 -0.200 1
ion Pt 0.50 0.000 -0.200 1
ion Pt 0.750 0.250 -0.100 1 
ion Pt 0.250 0.750 -0.100 1
ion Pt 0.250 0.250 -0.100 1
ion Pt 0.750 0.750 -0.100 1
ion Pt 0.50 0.000 0.000 1
ion Pt 0.000 0.000 0.000 1 
ion Pt 0.000 0.50 0.000 1 
ion Pt 0.500 0.50 0.000 1 
ion Pt 0.750 0.250 0.100 1 
ion Pt 0.250 0.750 0.100 1
ion Pt 0.250 0.250 0.100 1
ion Pt 0.750 0.750 0.100 1
ion Pt 0.000 0.000 0.200 1
ion Pt 0.000 0.50 0.200 1
ion Pt 0.50 0.000 0.200 1
ion Pt 0.50 0.50 0.200 1

In [91]:
np.array([-0.2,-0.1,0,.1,.2])*5/7

array([-0.14285714, -0.07142857,  0.        ,  0.07142857,  0.14285714])

ion Pt 0.000 0.000 -0.1428 1
ion Pt 0.500 0.500 -0.1428 1
ion Pt 0.000 0.50 -0.1428 1
ion Pt 0.50 0.000 -0.1428 1
ion Pt 0.750 0.250 -0.0714 1 
ion Pt 0.250 0.750 -0.0714 1
ion Pt 0.250 0.250 -0.0714 1
ion Pt 0.750 0.750 -0.0714 1
ion Pt 0.50 0.000 0.000 1
ion Pt 0.000 0.000 0.000 1 
ion Pt 0.000 0.50 0.000 1 
ion Pt 0.500 0.50 0.000 1 
ion Pt 0.750 0.250 0.0714 1 
ion Pt 0.250 0.750 0.0714 1
ion Pt 0.250 0.250 0.0714 1
ion Pt 0.750 0.750 0.0714 1
ion Pt 0.000 0.000 0.1428 1
ion Pt 0.000 0.50 0.1428 1
ion Pt 0.50 0.000 0.1428 1
ion Pt 0.50 0.50 0.1428 1

In [92]:
#for six layer unit cell
np.array([-0.187130,-0.093565,0.000000,0.093565,0.187130])*5/6

array([-0.15594167, -0.07797083,  0.        ,  0.07797083,  0.15594167])

ion Pt 0.000 0.000 -0.1667 1
ion Pt 0.500 0.500 -0.1667 1
ion Pt 0.000 0.50 -0.1667 1
ion Pt 0.50 0.000 -0.1667 1
ion Pt 0.750 0.250 -0.0833 1 
ion Pt 0.250 0.750 -0.0833 1
ion Pt 0.250 0.250 -0.0833 1
ion Pt 0.750 0.750 -0.0833 1
ion Pt 0.50 0.000 0.000 1
ion Pt 0.000 0.000 0.000 1 
ion Pt 0.000 0.50 0.000 1 
ion Pt 0.500 0.50 0.000 1 
ion Pt 0.750 0.250 0.0833 1 
ion Pt 0.250 0.750 0.0833 1
ion Pt 0.250 0.250 0.0833 1
ion Pt 0.750 0.750 0.0833 1
ion Pt 0.000 0.000 0.1428 1
ion Pt 0.000 0.50 0.1667 1
ion Pt 0.50 0.000 0.1667 1
ion Pt 0.50 0.50 0.1667 1

In [93]:
#for the carbon monoxide attached to the surface
OCdist=1.12/a_Pt_ang3
print(OCdist+0.282 )


0.33148716593053934


In [94]:
OCdist

0.04948716593053936

In [95]:
a_Pt_ang3


22.632130552233328

In [96]:
Ptpos=loadtxt('fracpos.dat')
print(Ptpos/2, size(Ptpos)/3)

[[ 0.3333335  0.1666665  0.262324 ]
 [ 0.3333335  0.6666665  0.262324 ]
 [ 0.8333335  0.1666665  0.262324 ]
 [ 0.8333335  0.6666665  0.262324 ]
 [-0.        -0.         0.321743 ]
 [ 0.         0.5        0.321743 ]
 [ 0.         1.         0.321743 ]
 [ 0.5       -0.         0.321743 ]
 [ 0.5        0.5        0.321743 ]
 [ 0.5        1.         0.321743 ]
 [ 1.        -0.         0.321743 ]
 [ 1.         0.5        0.321743 ]
 [ 1.         1.         0.321743 ]
 [ 0.1666665  0.3333335  0.381162 ]
 [ 0.1666665  0.8333335  0.381162 ]
 [ 0.6666665  0.3333335  0.381162 ]
 [ 0.6666665  0.8333335  0.381162 ]
 [ 0.3333335  0.1666665  0.440581 ]
 [ 0.3333335  0.6666665  0.440581 ]
 [ 0.8333335  0.1666665  0.440581 ]
 [ 0.8333335  0.6666665  0.440581 ]
 [ 0.         0.         0.       ]
 [-0.        -0.         0.5      ]
 [ 0.         0.5        0.       ]
 [ 0.         0.5        0.5      ]
 [ 0.         1.         0.       ]
 [ 0.         1.         0.5      ]
 [ 0.5        0.         0. 

In [97]:
Ptpos=(dot(bmat(diag([1/2,1/2,1])),Ptpos.T)).T

In [98]:
postPt=[]
for i in Ptpos:
    
    if 1. in i:
        print(i)
    else:
        postPt.append(array(i)[0])

[[0.       1.       0.643486]]
[[0.5      1.       0.643486]]
[[1.       0.       0.643486]]
[[1.       0.5      0.643486]]
[[1.       1.       0.643486]]
[[0. 0. 1.]]
[[0.  0.5 1. ]]
[[0. 1. 0.]]
[[0. 1. 1.]]
[[0.5 0.  1. ]]
[[0.5 0.5 1. ]]
[[0.5 1.  0. ]]
[[0.5 1.  1. ]]
[[1. 0. 0.]]
[[1. 0. 1.]]
[[1.  0.5 0. ]]
[[1.  0.5 1. ]]
[[1. 1. 0.]]
[[1. 1. 1.]]


In [99]:
for i in range(len(postPt)):
    print('ion Pt', end=' ' )
    for j in range(size(postPt[i])):
        print("%.4f" % postPt[i][j], end = ' ')
    print('1 \n')

ion Pt 0.3333 0.1667 0.5246 1 

ion Pt 0.3333 0.6667 0.5246 1 

ion Pt 0.8333 0.1667 0.5246 1 

ion Pt 0.8333 0.6667 0.5246 1 

ion Pt 0.0000 0.0000 0.6435 1 

ion Pt 0.0000 0.5000 0.6435 1 

ion Pt 0.5000 0.0000 0.6435 1 

ion Pt 0.5000 0.5000 0.6435 1 

ion Pt 0.1667 0.3333 0.7623 1 

ion Pt 0.1667 0.8333 0.7623 1 

ion Pt 0.6667 0.3333 0.7623 1 

ion Pt 0.6667 0.8333 0.7623 1 

ion Pt 0.3333 0.1667 0.8812 1 

ion Pt 0.3333 0.6667 0.8812 1 

ion Pt 0.8333 0.1667 0.8812 1 

ion Pt 0.8333 0.6667 0.8812 1 

ion Pt 0.0000 0.0000 0.0000 1 

ion Pt 0.0000 0.5000 0.0000 1 

ion Pt 0.5000 0.0000 0.0000 1 

ion Pt 0.5000 0.5000 0.0000 1 



In case shit hits the fan

ion Pt -0.333333  0.333333 -0.475352   1
ion Pt  0.000000  0.000000 -0.356514   1
ion Pt  0.333333 -0.333333 -0.237676   1
ion Pt -0.333333  0.333333 -0.118838   1
ion Pt  0.000000  0.000000  0.000000   1

In [100]:
import statistics
postPt=array(postPt)
postPt[-4:,2]=postPt[-4:,2]+1
centval=statistics.median(array(postPt)[:,2])
postPt[:,2]=postPt[:,2]-centval

Ptcoords=(array(dot(diag([1,1,floor((5*(sqrt(3)*a_Pt_ang/3))*1.88973 +15)/(a_Pt_ang3*1.88973)]),mat(postPt).T).T)) 
#Ptcoords=list(array(dot(diag([1,1,1]),mat(postPt).T).T)) 
centval=statistics.median((Ptcoords)[:,2])
#Ptcoords[:,2]=Ptcoords[:,2]-max((Ptcoords)[:,2])
## the weird expression 
#in the denominator is shankars choice 
#for cleaving the surface, leaving a 15 bohr separation between slabs and 5 layers of thickness sqrt3*a/3 each
## I work first in angstrom and that is why I put the 1.88973


In [101]:
Ptcoords

array([[ 0.3333335 ,  0.1666665 , -0.20006109],
       [ 0.3333335 ,  0.6666665 , -0.20006109],
       [ 0.8333335 ,  0.1666665 , -0.20006109],
       [ 0.8333335 ,  0.6666665 , -0.20006109],
       [ 0.        ,  0.        , -0.10003055],
       [ 0.        ,  0.5       , -0.10003055],
       [ 0.5       ,  0.        , -0.10003055],
       [ 0.5       ,  0.5       , -0.10003055],
       [ 0.1666665 ,  0.3333335 ,  0.        ],
       [ 0.1666665 ,  0.8333335 ,  0.        ],
       [ 0.6666665 ,  0.3333335 ,  0.        ],
       [ 0.6666665 ,  0.8333335 ,  0.        ],
       [ 0.3333335 ,  0.1666665 ,  0.10003055],
       [ 0.3333335 ,  0.6666665 ,  0.10003055],
       [ 0.8333335 ,  0.1666665 ,  0.10003055],
       [ 0.8333335 ,  0.6666665 ,  0.10003055],
       [ 0.        ,  0.        ,  0.20006109],
       [ 0.        ,  0.5       ,  0.20006109],
       [ 0.5       ,  0.        ,  0.20006109],
       [ 0.5       ,  0.5       ,  0.20006109]])

In [102]:
for i in range(len(Ptcoords)):
    print('ion Pt', end=' ' )
    for j in range(size(Ptcoords[i])):
        print("%.4f" % Ptcoords[i][j], end = ' ')
    print('1 ')

ion Pt 0.3333 0.1667 -0.2001 1 
ion Pt 0.3333 0.6667 -0.2001 1 
ion Pt 0.8333 0.1667 -0.2001 1 
ion Pt 0.8333 0.6667 -0.2001 1 
ion Pt 0.0000 0.0000 -0.1000 1 
ion Pt 0.0000 0.5000 -0.1000 1 
ion Pt 0.5000 0.0000 -0.1000 1 
ion Pt 0.5000 0.5000 -0.1000 1 
ion Pt 0.1667 0.3333 0.0000 1 
ion Pt 0.1667 0.8333 0.0000 1 
ion Pt 0.6667 0.3333 0.0000 1 
ion Pt 0.6667 0.8333 0.0000 1 
ion Pt 0.3333 0.1667 0.1000 1 
ion Pt 0.3333 0.6667 0.1000 1 
ion Pt 0.8333 0.1667 0.1000 1 
ion Pt 0.8333 0.6667 0.1000 1 
ion Pt 0.0000 0.0000 0.2001 1 
ion Pt 0.0000 0.5000 0.2001 1 
ion Pt 0.5000 0.0000 0.2001 1 
ion Pt 0.5000 0.5000 0.2001 1 


In [103]:
centval

0.0